src/clusterd/src/usage_metrics.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! Support for collecting system usage metrics.
+//!
+//! Currently only disk usage is supported. We may want to add CPU and memory usage in the future.
+
+use std::path::PathBuf;
+
+use serde::Serialize;
+use tracing::error;
+
+/// A system usage metrics collector.
+pub(crate) struct Collector {
+    pub disk_root: Option<PathBuf>,
+}
+
+impl Collector {
+    /// Collect current system usage metrics.
+    pub fn collect(&self) -> Usage {
+        Usage {
+            disk_bytes: self.collect_disk_usage(),
+        }
+    }
+
+    fn collect_disk_usage(&self) -> Option<u64> {
+        let Some(root) = &self.disk_root else {
+            return None;
+        };
+
+        let stat = match nix::sys::statvfs::statvfs(root) {
+            Ok(stat) => stat,
+            Err(err) => {
+                error!("statvfs error: {err}");
+                return None;
+            }
+        };
+
+        // `fsblkcnt_t` is a `u32` on macOS but a `u64` on Linux.
+        #[allow(clippy::useless_conversion)]
+        let used_blocks = u64::from(stat.blocks() - stat.blocks_free());
```

Comment:
Something to discuss: [statvfs](https://man7.org/linux/man-pages/man3/statvfs.3.html) returns two different free block values:
* `f_bfree`: Number of free blocks
* `f_bavail`: Number of free blocks for unprivileged users

We could use `blocks_available()` here instead of `blocks_free()`. Apparently the space reserved for root is 5% by default, but that might not be the case in our k8s setup.

Reply:
Looks like in cloud we only reserve 16MiB, so it doesn't really matter either way:

```
materialize@mz6tylgpv9fe-cluster-u1-replica-u1-gen-2-0:/$ df /scratch
Filesystem                                                                    1K-blocks    Used Available Use% Mounted on
/dev/mapper/instance--store--vg-pvc--e3579d60--2fe8--48c2--94f3--eec2b60c4155   8154588 1334568   6803636  17% /scratch
```

I'll change to available because of Alex' vote and because it's consistent with `df`.